pkg/certs/cert-inspection/certgraphanalysis/collector.go

```diff
@@ -216,23 +244,73 @@ func MergePKILists(ctx context.Context, first, second *certgraphapi.PKIList) *ce
 	certList := &certgraphapi.CertKeyPairList{
 		Items: append(first.CertKeyPairs.Items, second.CertKeyPairs.Items...),
 	}
-	certList = deduplicateCertKeyPairList(certList)
 
-	caBundlesList := &certgraphapi.CertificateAuthorityBundleList{
+	caBundleList := &certgraphapi.CertificateAuthorityBundleList{
 		Items: append(first.CertificateAuthorityBundles.Items, second.CertificateAuthorityBundles.Items...),
 	}
-	caBundlesList = deduplicateCABundlesList(caBundlesList)
 
 	inClusterData := certgraphapi.PerInClusterResourceData{}
 	inClusterData.CertKeyPairs = append(inClusterData.CertKeyPairs, first.InClusterResourceData.CertKeyPairs...)
 	inClusterData.CertKeyPairs = append(inClusterData.CertKeyPairs, second.InClusterResourceData.CertKeyPairs...)
 	inClusterData.CertificateAuthorityBundles = append(inClusterData.CertificateAuthorityBundles, first.InClusterResourceData.CertificateAuthorityBundles...)
 	inClusterData.CertificateAuthorityBundles = append(inClusterData.CertificateAuthorityBundles, second.InClusterResourceData.CertificateAuthorityBundles...)
-	//TODO[vrutkovs] deduplicate inClusterData?
+
+	onDiskResourceData := &certgraphapi.PerOnDiskResourceData{
+		TLSArtifact: append(first.OnDiskResourceData.TLSArtifact, second.OnDiskResourceData.TLSArtifact...),
+	}
+
+	param := pathUpdateParams{
+		prefix:   prefix,
+		nodeList: nodeList,
+	}
+	stripPrefixFromPaths(certList, caBundleList, onDiskResourceData, param,
```

Comment:
Why did this move from when we collect data to when we merge it?  Doing it at collection adjusts how every layer in the stack sees the data. Versus doing it here where it means other locations in the code see the old names.

Reply:
That would require all collection layers to be aware of rootfsmount (doable) and require node list permissions. Its implementable, but that makes us dedup and filter twice - once on the node and once for incluster resources

Reply:
Let's push  it into the collection path.